spring-thrift-starter/src/main/java/info/developerblog/spring/thrift/client/ThriftClientBeanPostProcessor.java

```diff
@@ -109,6 +113,16 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
         return bean;
     }
 
+    //We have to get a real bean in order to inject a thrift client into the bean instead of its proxy.
+    @SneakyThrows
+    private Object getTargetBean(Object bean) {
+        Object target = bean;
+        while (AopUtils.isAopProxy(target)) {
```

Comment:
Technically, isAopProxy method is quite flexible and it can determine proxies, which are created either by spring or by pure JDK Proxy and CGLIB.

I haven't understood which kind of a test you want to see: the test which proves that we catch NPE with Async with current code or the test which proves that we handle any kind of proxies correctly.